### DISCUSSION TITLE
Efficient fine-tuning using Adapters.

### DISCUSSION THREAD

**Discussion by PSanni** [coqui-ai/TTS#3382] [General Q&A]
Full finetuning of the model is resource hungry in terms of compute and data. What would be the steps to integrate support for adapters such as LoRa ?

**Comment by erogol** [comment]
(Qs to discussions) 

yes possible. We internally tried such and they work well with lower resources but the best was training the whole model.

**Comment by PSanni** [comment]
What modifications did you implement to enable it? Which specific modules did you target in the LoRa ? Could you provide a some overview of the changes made?

**Comment by CrackerHax** [comment]
Would be interested in this as well since there is no documentation of how to train the labelled "speaker_id" models (or at least how to use them once trained). I was able to train a labeled dataset but trying to load the model using TTS(model), (which is the only way to apparently set your speaker_id) I get the error: "KeyError: 'run'". So yeah if labelled speakers are unusable then adapters are the next best thing if you are memory-swapping and need multiple loaded voices.
